tests/unit/staff/views/test_users.py

```diff
@@ -484,102 +484,119 @@ def test_userlist_filter_by_invalid_role(rf, core_developer):
 
 
 def test_userlist_filter_by_any_roles_yes_includes_global_roles(rf, core_developer):
-    UserFactory(roles=[OutputPublisher])
-    UserFactory(roles=[])
+    user = UserFactory(roles=[OutputPublisher])
+    UserFactory()
 
     request = rf.get("/?any_roles=yes")
     request.user = core_developer
 
     response = UserList.as_view()(request)
 
-    assert (
-        len(response.context_data["object_list"]) == 2
-    )  # this includes the core_developer
+    assert set_from_qs(response.context_data["object_list"]) == {
```

Comment:
Thank you for converting these to use `set_from_qs` it's much easier to see what's going on now, especially with the extra assertions in later tests.